src/parser/mod.rs

```diff
@@ -6770,7 +6942,19 @@ impl<'a> Parser<'a> {
                 Keyword::JSON => Ok(DataType::JSON),
                 Keyword::JSONB => Ok(DataType::JSONB),
                 Keyword::REGCLASS => Ok(DataType::Regclass),
-                Keyword::STRING => Ok(DataType::String(self.parse_optional_precision()?)),
+                Keyword::STRING => {
+                    if dialect_of!(self is ClickHouseDialect) {
+                        Ok(DataType::String(None))
+                    } else {
+                        Ok(DataType::String(self.parse_optional_precision()?))
+                    }
```

Comment:
is the intention here solely to have the parser reject e.g. `foo STRING(42)` as invalid input? if so then its probably not necessary, we can have the parser accept it and keep the codepath simpler given that's slightly towards semantics

Reply:
Reverted.